internal/xds/translator/listener_ready.go

```diff
@@ -0,0 +1,104 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package translator
+
+import (
+	"fmt"
+
+	corev3 "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
+	listenerv3 "github.com/envoyproxy/go-control-plane/envoy/config/listener/v3"
+	routev3 "github.com/envoyproxy/go-control-plane/envoy/config/route/v3"
+	hcmv3 "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/network/http_connection_manager/v3"
+	"github.com/envoyproxy/go-control-plane/pkg/wellknown"
+
+	egv1a1 "github.com/envoyproxy/gateway/api/v1alpha1"
+	"github.com/envoyproxy/gateway/internal/ir"
+	"github.com/envoyproxy/gateway/internal/utils/protocov"
+	"github.com/envoyproxy/gateway/internal/xds/bootstrap"
+	"github.com/envoyproxy/gateway/internal/xds/filters"
+)
+
+func buildReadyListener(ready *ir.ReadyListener) (*listenerv3.Listener, error) {
+	ipv4Compact := false
+	if ready.IPFamily == egv1a1.IPv6 || ready.IPFamily == egv1a1.DualStack {
+		ipv4Compact = true
+	}
+	hcmFilters := make([]*hcmv3.HttpFilter, 0, 3)
+	healthcheckFilter, err := filters.GenerateHealthCheckFilter(bootstrap.EnvoyReadinessPath)
+	if err != nil {
+		return nil, err
+	}
+	hcmFilters = append(hcmFilters, healthcheckFilter)
+
+	router, err := filters.GenerateRouterFilter(false)
+	if err != nil {
+		return nil, err
+	}
+	hcmFilters = append(hcmFilters, router)
+
+	hcm := &hcmv3.HttpConnectionManager{
+		StatPrefix: "eg-ready-http",
+		RouteSpecifier: &hcmv3.HttpConnectionManager_RouteConfig{
+			RouteConfig: &routev3.RouteConfiguration{
+				Name: "ready_route",
+				VirtualHosts: []*routev3.VirtualHost{
+					{
+						Name:    "ready_route",
```

Comment:
is the route needed ? my worry is, the response is not based off health

Reply:
we cannot leave route_config nil, or I missed something?

Reply:
can we return `500` in the DR then ? wanna make sure only the good case works

Reply:
updated, hope CI happy.